truffle/src/main/java/org/jruby/truffle/nodes/ext/BigDecimalNodes.java

```diff
@@ -178,175 +217,327 @@ public static boolean isNan(RubyBasicObject value) {
             return getBigDecimalType(value) == Type.NAN;
         }
 
+        private void setupCreateBigDecimal() {
+            if (createBigDecimal == null)
+                createBigDecimal = insert(CreateBigDecimalNodeFactory.create(getContext(), getSourceSection(), null, null));
+        }
+
+        protected RubyBasicObject createBigDecimal(VirtualFrame frame, Object value) {
+            setupCreateBigDecimal();
+            return createBigDecimal.executeCreate(frame, value);
+        }
+
+        protected RubyBasicObject createBigDecimal(VirtualFrame frame, Object value, RubyBasicObject self) {
+            setupCreateBigDecimal();
+            return createBigDecimal.executeCreate(frame, value, self);
+        }
+
+        private void setupLimitCall() {
+            if (limitCall == null)
+                limitCall = insert(DispatchHeadNodeFactory.createMethodCall(getContext()));
+        }
+
+        private void setupLimitIntegerCast() {
+            if (limitIntegerCast == null)
+                limitIntegerCast = insert(IntegerCastNodeGen.create(getContext(), getSourceSection(), null));
+        }
+
+        protected int getLimit(VirtualFrame frame) {
+            setupLimitCall();
+            setupLimitIntegerCast();
+
+            return limitIntegerCast.executeInteger(frame, limitCall.call(frame, getBigDecimalClass(), "limit", null));
+        }
+
+        private void setupRoundModeCall() {
+            if (roundModeCall == null)
+                roundModeCall = insert(DispatchHeadNodeFactory.createMethodCall(getContext()));
+        }
+
+        private void setupRoundModeIntegerCast() {
+            if (roundModeIntegerCast == null)
+                roundModeIntegerCast = insert(IntegerCastNodeGen.create(getContext(), getSourceSection(), null));
+        }
+
+        protected RoundingMode getRoundMode(VirtualFrame frame) {
+            setupRoundModeCall();
+            setupRoundModeIntegerCast();
+
+            return toRoundingMode(roundModeIntegerCast.executeInteger(frame,
+                    // TODO (pitr 21-Jun-2015): read the actual constant
+                    roundModeCall.call(frame, getBigDecimalClass(), "mode", null, 256)));
+        }
+
+        @Deprecated
```

Comment:
yeah it'll be removed soon when I fix `<=>` not to use it. I was just for me so I see the call easier.